packages/admin-ui/ui/src/domain/discounts/details/configurations/edit-configurations.tsx

```diff
@@ -1,205 +1,79 @@
 import { Discount } from "@medusajs/medusa"
 import { useAdminUpdateDiscount } from "medusa-react"
 import React, { useEffect } from "react"
-import { Controller, useForm } from "react-hook-form"
-import DatePicker from "../../../../components/atoms/date-picker/date-picker"
-import TimePicker from "../../../../components/atoms/date-picker/time-picker"
+import { useForm } from "react-hook-form"
+import DiscountConfigurationForm, {
+  DiscountConfigurationFormType,
+} from "../../../../components/forms/discount/discount-configuration-form"
 import Button from "../../../../components/fundamentals/button"
-import AvailabilityDuration from "../../../../components/molecules/availability-duration"
-import InputField from "../../../../components/molecules/input"
 import Modal from "../../../../components/molecules/modal"
-import SwitchableItem from "../../../../components/molecules/switchable-item"
 import useNotification from "../../../../hooks/use-notification"
 import { getErrorMessage } from "../../../../utils/error-messages"
+import { nestedForm } from "../../../../utils/nested-form"
 
 type EditConfigurationsProps = {
   discount: Discount
   onClose: () => void
+  open: boolean
 }
 
 type ConfigurationsForm = {
-  starts_at: Date | null
-  ends_at: Date | null
-  usage_limit: number | null
-  valid_duration: string | null
+  config: DiscountConfigurationFormType
 }
 
 const EditConfigurations: React.FC<EditConfigurationsProps> = ({
   discount,
   onClose,
+  open,
 }) => {
   const { mutate, isLoading } = useAdminUpdateDiscount(discount.id)
   const notification = useNotification()
 
-  const { control, handleSubmit, reset } = useForm<ConfigurationsForm>({
-    defaultValues: mapConfigurations(discount),
+  const form = useForm<ConfigurationsForm>({
+    defaultValues: getDefaultValues(discount),
   })
 
-  const onSubmit = (data: ConfigurationsForm) => {
+  const { handleSubmit, reset } = form
+
+  const onSubmit = handleSubmit((data) => {
     mutate(
       {
-        starts_at: data.starts_at ?? new Date(),
-        ends_at: data.ends_at,
+        starts_at: data.config.starts_at ?? new Date(),
+        ends_at: data.config.ends_at,
         usage_limit:
-          data.usage_limit && data.usage_limit > 0 ? data.usage_limit : null,
-        valid_duration: data.valid_duration,
+          data.config.usage_limit && data.config.usage_limit > 0
```

Comment:
```suggestion
          data.config?.usage_limit > 0
```

Reply:
Can't do that in TS since it might be null and you can check null > 0. I would then have to do  (data.config?.usage_limit || 0) > 0. We should clean up the type of this, but since we will be revisting all domains soon I think we should let it be for now :D

Reply:
Since we are looking for a boolean and check that usage_limit if present is greater than 0. Wouldn’t !!data.config.usage_limit be enought?